extensions/extension-editing/src/extensionLinter.ts

```diff
@@ -145,6 +147,19 @@ export class ExtensionLinter {
 							}
 						}
 					}
+					const activationEventsNode = findNodeAtLocation(tree, ['activationEvents']);
```

Comment:
Maybe I am not seeing it but I am missing the checks that the respective commands and notebooks are actually contributed by that extension. E.g we don't want to warn when an extension does bystander-activation on certain commands or notebooks, right?

Reply:
Good point, maybe we should not lint on all contributions then, because that way we would have to duplicate the logic for parsing contributions into this extension. 

At minimum though I think we should emit an error if an extension tries to use `onNotebookSerializer` and there's no corresponding notebooks contribution, because extensions should not be allowed to do bystander activation using `onNotebookSerializer`, instead they should be using `onNotebook`. What do you think?

Reply:
> because extensions should not be allowed to do bystander activation using onNotebookSerializer, instead they should be using onNotebook. What do you think?

I like that. We could also be super strict and always ignore explict `onNotebookSerializer` and make it implicit only

Reply:
Yeah that sounds good although we would have to special case this event in extension scanning land which is a bit ugly. Shall I update this PR to just emit an error for explicit usage of `onNotebookSerializer`? Was also thinking of emitting a vsce error but that might be overkill.

Reply:
> Shall I update this PR to just emit an error for explicit usage of onNotebookSerializer?

Yeah, let's do that

Reply:
✅